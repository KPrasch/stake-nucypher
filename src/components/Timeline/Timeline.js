import React, { Fragment } from 'react';
import classnames from 'classnames';
import './Timeline.scss';

function dateFormat(dateNum) {
  const date = new Date(dateNum * 1000);
  const options = { month: 'numeric', day: 'numeric', year: 'numeric' };
  return date.toLocaleDateString('en-US', options);
}

const now = Date.now() / 1000;

export default function Timeline(props) {
  if (props.timelinePoints) {
    return <>
      <ul className="timeline" id="timeline">
        {
          props.timelinePoints.map((point, index) => <Fragment key={index}>
            <li className={classnames('li', { 'complete': (point.date <= now || props.completedIndex >= index )})} style={{width: 100/props.timelinePoints.length + '%'}}>
              <div className="timeline">
                {
                  point.date ? <span className="date">{ dateFormat(point.date) }</span> : null
                }
                {
                  point.textPoint ? <span className="text-point">{ point.textPoint }</span> : null
                }
              </div>
              <div className="status">
                {
                  point.label ? <h6>{point.label}</h6> : null
                }
              </div>
            </li>
          </Fragment>)
        }
      </ul>
    </>;
  } else {
    return null;
  }
}
