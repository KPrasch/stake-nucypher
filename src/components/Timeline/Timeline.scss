.timeline {
  list-style-type: none;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;

  .date {
    padding-bottom: 20px;
    min-height: 41px;
  }
  .text-point {
    padding-bottom: 20px;
    min-height: 41px;
  }
}
.li {
  transition: all 200ms ease-in;
}

.timestamp {
  margin-bottom: 20px;
  padding: 0px 40px;
  display: flex;
  flex-direction: column;
  align-items: center;
  font-weight: 100;
}
.status {
  padding: 0px 40px;
  display: flex;
  justify-content: center;
  border-top: 2px solid #D6DCE0;
  position: relative;
  transition: all 200ms ease-in;
  padding-top: 15px;
  min-height: 82px;
  text-align: center;

  h4 {
    font-weight: 600;
  }

  h6 {
    display: inline-block;
    position: absolute;
    left: 80%;
  }

  &:before {
    content: '';
    width: 25px;
    height: 25px;
    background-color: white;
    border-radius: 25px;
    border: 1px solid #ddd;
    position: absolute;
    top: -15px;
    left: 42%;
    transition: all 200ms ease-in;
  }
}
.li.complete {
  .status {
    border-top: 2px solid #007bff;
    &:before {
      background-color: #007bff;
      border: none;
      transition: all 200ms ease-in;
    }
    h5 {
      color: #007bff;
    }
  }
}

@media (min-device-width: 320px) and (max-device-width: 700px) {
  .timeline {
    list-style-type: none;
    display: block;
  }
  .li {
    transition: all 200ms ease-in;
    display: flex;
    width: inherit;
  }
  .timestamp {
    width: 100px;
  }
  .status {
    &:before {
      left: -8%;
      top: 30%;
      transition: all 200ms ease-in;
    }
  }
}
