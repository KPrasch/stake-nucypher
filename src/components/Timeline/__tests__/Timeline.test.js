import React from 'react';
import Timeline from '../Timeline';
import { render } from '@testing-library/react'

describe('<Timeline />', () => {
  const timelinePoints = [
    {
      date: '1589459576',
      label: 'Escrow phase'
    },
    {
      date: '1589459576',
      label: 'Escrow cancellation window'
    },
    {
      date: '1589459576',
      label: 'Claiming tokens window'
    },
    {
      textPoint: 'Stake creation',
      label: 'Running node'
    },
    {
      textPoint: 'Ether claimed'
    }
  ];

  it('should render correctly', () => {
    const { asFragment } = render(<Timeline timelinePoints={timelinePoints} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
