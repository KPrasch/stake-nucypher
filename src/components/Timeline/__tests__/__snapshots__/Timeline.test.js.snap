// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timeline /> should render correctly 1`] = `
<DocumentFragment>
  <ul
    class="timeline"
    id="timeline"
  >
    <li
      class="li complete"
      style="width: 20%;"
    >
      <div
        class="timeline"
      >
        <span
          class="date"
        >
          5/14/2020
        </span>
      </div>
      <div
        class="status"
      >
        <h6>
          Escrow phase
        </h6>
      </div>
    </li>
    <li
      class="li complete"
      style="width: 20%;"
    >
      <div
        class="timeline"
      >
        <span
          class="date"
        >
          5/14/2020
        </span>
      </div>
      <div
        class="status"
      >
        <h6>
          Bids cancelation window
        </h6>
      </div>
    </li>
    <li
      class="li complete"
      style="width: 20%;"
    >
      <div
        class="timeline"
      >
        <span
          class="date"
        >
          5/14/2020
        </span>
      </div>
      <div
        class="status"
      >
        <h6>
          Claiming tokens window
        </h6>
      </div>
    </li>
    <li
      class="li"
      style="width: 20%;"
    >
      <div
        class="timeline"
      >
        <span
          class="text-point"
        >
          Stake creation
        </span>
      </div>
      <div
        class="status"
      >
        <h6>
          Running node
        </h6>
      </div>
    </li>
    <li
      class="li"
      style="width: 20%;"
    >
      <div
        class="timeline"
      >
        <span
          class="text-point"
        >
          Ether claimed
        </span>
      </div>
      <div
        class="status"
      />
    </li>
  </ul>
</DocumentFragment>
`;
